src/base-client.ts

```diff
@@ -1,169 +1,186 @@
+                xhr.onreadystatechange = () => {
+                    if (xhr.readyState === 4) {
+                        if (xhr.status === 200) {
+                            resolve(xhr.responseText);
+                        } else {
+                            reject(ErrorResponse.fromXHR(xhr));
+                        }
+                    }
+                };
+                
+                xhr.send(JSON.stringify(options.body));
+            });
+        }
 
-    newResumableSubscription(subOptions: ResumableSubscribeOptions): ResumableSubscription {
-        return new ResumableSubscription(
-            () => {
-                return this.createXHR(this.baseURL, {
-                    method: "SUBSCRIBE",
-                    path: subOptions.path,
-                    headers: {},
-                    body: null,
+        newStatelessSubscription(subOptions: StatelessSubscribeOptions): StatelessSubscription {
+            const method = "SUBSCRIBE";
+            if( !subOptions.retryStrategy ) {
+                subOptions.retryStrategy = new ExponentialBackoffRetryStrategy({
+                    logger: this.logger,
+                    requestMethod: method
                 });
-            },
-            subOptions
-        );
-    }
-
-    private createXHR(baseURL: string, options: RequestOptions): XMLHttpRequest {
-        let XMLHttpRequest: any = this.XMLHttpRequest;
-        let xhr = new XMLHttpRequest();
-        let path = options.path.replace(/^\/+/, "");
-        let endpoint = `${baseURL}/${path}`;
-
-        xhr.open(options.method.toUpperCase(), endpoint, true);
-
-        if (options.body) {
-            xhr.setRequestHeader("content-type", "application/json");
+            }
+            return new StatelessSubscription(
+                () => { 
+                    return this.createXHR(this.baseURL, {
+                        method: method,
+                        path: subOptions.path,
+                        headers: {},
+                        body: null,
+                    });
+                },
+                subOptions
+            );
         }
-
-        if (options.jwt) {
-            xhr.setRequestHeader("authorization", `Bearer ${options.jwt}`);
+        
+        newResumableSubscription(subOptions: ResumableSubscribeOptions): ResumableSubscription {
+            const method = "SUBSCRIBE";
+            if( !subOptions.retryStrategy ) {
+                subOptions.retryStrategy = new ExponentialBackoffRetryStrategy({
+                    logger: this.logger,
+                    requestMethod: method
+                });
+            }
+            return new ResumableSubscription(
+                () => {
+                    return this.createXHR(this.baseURL, {
+                        method: method,
+                        path: subOptions.path,
+                        headers: {},
+                        body: null,
+                    });
+                },
```

Comment:
I think you're leaking abstractions here - resumable subscriptions should not know about XHR. They don't need an XHR constructor - they need a subscription constructor instead.